components/match2/wikis/ageofempires/match_group_input_custom.lua

```diff
@@ -0,0 +1,486 @@
+---
+-- @Liquipedia
+-- wiki=ageofempires
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+local Array = require('Module:Array')
+local Faction = require('Module:Faction')
+local Game = require('Module:Game')
+local Json = require('Module:Json')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local String = require('Module:StringUtils')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local MatchGroupInput = Lua.import('Module:MatchGroup/Input')
+local Opponent = Lua.import('Module:Opponent')
+local Streams = Lua.import('Module:Links/Stream')
+
+local ALLOWED_STATUSES = {'W', 'FF', 'DQ', 'L'}
+local CONVERT_STATUS_INPUT = {W = 'W', FF = 'FF', L = 'L', DQ = 'DQ', ['-'] = 'L'}
+local DEFAULT_LOSS_STATUSES = {'FF', 'L', 'DQ'}
+local MAX_NUM_OPPONENTS = 2
+
+local CustomMatchGroupInput = {}
+
+--- called from Module:MatchGroup
+---@param match table
+---@param options table?
+---@return table
+function CustomMatchGroupInput.processMatch(match, options)
+	assert(not Logic.readBool(match.ffa), 'FFA is not yet supported in AoE match2.')
+	mw.logObject(match, 'Match')
+	Table.mergeInto(match, CustomMatchGroupInput._readDate(match))
+	CustomMatchGroupInput._getOpponents(match)
+	CustomMatchGroupInput._getTournamentVars(match)
+	CustomMatchGroupInput._processMaps(match)
+	CustomMatchGroupInput._calculateWinner(match)
+	CustomMatchGroupInput._updateFinished(match)
+	match.stream = Streams.processStreams(match)
+	CustomMatchGroupInput._getLinks(match)
+	CustomMatchGroupInput._getVod(match)
+	CustomMatchGroupInput._getExtraData(match)
+	mw.logObject(match, 'ProcessedMatch')
+	return match
+end
+
+---@param matchArgs table
+function CustomMatchGroupInput._readDate(matchArgs)
+	return MatchGroupInput.readDate(matchArgs.date, {
+		'tournament_startdate',
+		'tournament_enddate',
+	})
+end
+
+---@param match table
+function CustomMatchGroupInput._getTournamentVars(match)
+	local failure
+	match.mapsInfo, failure = Json.parse(Variables.varDefault('tournament_maps'))
+	match.bestof = Logic.emptyOr(match.bestof, Variables.varDefault('bestof'))
+	Variables.varDefine('bestof', match.bestof)
+
+	if failure or Logic.isEmpty(match.mapsInfo) or String.isEmpty(match.game) then
+		-- likely in preview. Fetch from LPDB
+		local data = mw.ext.LiquipediaDB.lpdb('tournament', {
+			conditions = '[[pagename::' ..
```

Comment:
you might have to fetch it from parent instead of current page (e.g. if matches are on a subpage)
not sure if ape hdb sets the map info or not 
(am on phone)

Reply:
HDB sets the variable, not LPDB data, so it would work unless in preview.
Using parent is a good idea, forgot about that 👍🏼